datalad/distribution/dataset.py

```diff
@@ -99,6 +102,41 @@ def repo(self):
                 self._repo = AnnexRepo(self._repo.path, create=False)
         return self._repo
 
+    @property
+    def config(self):
+        """Get an instance of the parser for the persistent dataset configuration.
+
+        Returns
+        -------
+        GitConfigParser
+        """
+        if self._cfg is None:
+            cfgpath = opj(self.path, '.datalad', 'config')
+            if not exists(cfgpath):
+                return None
+            self._cfg = git.GitConfigParser(cfgpath, read_only=True)
+        return self._cfg
+
+    @property
+    def config_writer(self):
+        """Get an instance of the parser for modifying the persistent dataset
+        configuration.
+
+        Returns
+        -------
+        GitConfigParser
+        """
+        if self._cfg is None:
+            cfgpath = opj(self.path, '.datalad')
+            if not exists(cfgpath):
+                os.makedirs(cfgpath)
+            self._cfg = git.GitConfigParser(
+                opj(cfgpath, 'config'),
+                read_only=False,
+                merge_includes=False)
+        return self._cfg
```

Comment:
True, but this needs more planning. It is trivial to read config from many files, but not to decide which config to write to what file. I guess a dedicated issue, and a matching hangout agenda item is due....